src/timeseers/likelihood.py

```diff
@@ -4,29 +4,49 @@
 
 class Likelihood(ABC):
     """Subclasses should implement the observed method which defines an observed random variable"""
+    @property
+    def distribution(self):
+        raise NotImplementedError
+
+    def observed(self, mu, y_scaled, sample_weight):
+        model_parameters = self.define_priors()
+
+        model_parameters["mu"] = mu
+
+        if sample_weight is None:
+            self.distribution("obs", **model_parameters, observed=y_scaled)
+        else:
+            pm.Potential("obs", sample_weight * self.distribution.dist(**model_parameters).logp(y_scaled))
+
     @abstractmethod
-    def observed(self, mu, y_scaled):
+    def define_priors(self):
         pass
 
 
 class Gaussian(Likelihood):
     """Gaussian likelihood with constant variance"""
+    distribution = pm.Normal
+
     def __init__(self, sigma=0.5):
         self.sigma = sigma
 
-    def observed(self, mu, y_scaled):
-        sigma = pm.HalfCauchy("sigma", self.sigma)
-        pm.Normal("obs", mu=mu, sd=sigma, observed=y_scaled)
+    def define_priors(self):
```

Comment:
I'm not sure exactly why, but I'm leaning towards making this a private method and calling it `_likelihood_priors` instead.

Reply:
Took the _likelihood_priors route. I prefer the separation of the method and init, but let me know if you prefer the alternative